suite-native/coin-enabling/src/components/CoinEnablingForm.tsx

```diff
@@ -0,0 +1,94 @@
+import { type ComponentProps, useCallback, useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { changeCoinVisibility } from '@suite-common/wallet-core';
+import { useAlert } from '@suite-native/alerts';
+import { EventType, analytics } from '@suite-native/analytics';
+import {
+    selectDeviceEnabledDiscoveryNetworkSymbols,
+    selectDiscoveryNetworkSymbols,
+} from '@suite-native/discovery';
+import { Form, useForm } from '@suite-native/forms';
+import { Translation } from '@suite-native/intl';
+
+import { DiscoveryCoinsFilter } from './DiscoveryCoinsFilter';
+import { CoinEnablingFormValues, coinEnablingFormValidationSchema } from '../coinEnablingSchema';
+
+export const CoinEnablingForm = () => {
+    const dispatch = useDispatch();
+    const navigation = useNavigation();
+
+    const enabledNetworkSymbols = useSelector(selectDeviceEnabledDiscoveryNetworkSymbols);
+    const networkSymbols = useSelector(selectDiscoveryNetworkSymbols);
+
+    const { showAlert } = useAlert();
+
+    const showLastNetworkAlert = useCallback(
+        () =>
+            showAlert({
+                title: <Translation id="moduleSettings.coinEnabling.oneNetworkSymbolAlert.title" />,
+                description: (
+                    <Translation id="moduleSettings.coinEnabling.oneNetworkSymbolAlert.description" />
+                ),
+                primaryButtonTitle: <Translation id="generic.buttons.gotIt" />,
+                primaryButtonVariant: 'redBold',
+            }),
+        [showAlert],
+    );
+
+    const form = useForm<CoinEnablingFormValues>({
+        defaultValues: {
+            enabledCoins: enabledNetworkSymbols,
+        },
+        validation: coinEnablingFormValidationSchema,
+    });
+
+    const handleSubmit = form.handleSubmit(values => {
+        const changedCoins = networkSymbols.filter(
+            symbol =>
+                enabledNetworkSymbols.includes(symbol) !== values.enabledCoins.includes(symbol),
+        );
+
+        if (changedCoins.length === 0) return;
+
+        changedCoins.forEach(symbol => {
+            const isEnabled = values.enabledCoins.includes(symbol);
+            dispatch(changeCoinVisibility({ symbol, shouldBeVisible: isEnabled }));
+
+            analytics.report({
+                type: EventType.SettingsChangeCoinEnabled,
+                payload: {
+                    symbol,
+                    value: isEnabled,
+                },
+            });
+        });
+    });
+
+    useEffect(
+        () =>
+            navigation.addListener('beforeRemove', () => {
+                if (form.formState.isDirty) {
+                    handleSubmit();
+                }
+            }),
+        [navigation, form.formState.isDirty, handleSubmit],
+    );
```

Comment:
![image](https://github.com/user-attachments/assets/446fe775-14b9-4618-a436-a5f2854957ea)

Reference: https://react-hook-form.com/docs/useform/formstate

Ideally, destructure from form state like so: 
```
    const {
        handleSubmit,
        formState: { isDirty },
    } = useFormContext();
```

Reply:
Thanks, I wasn't aware of that at all! I think the the check for dirtiness is redundant here anyway, checking for changed networks in handleSubmit should be cheap, so I removed the condition completely to simplify it.